sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/MSIToken.java

```diff
@@ -72,6 +65,39 @@ public String getToken() {
         return accessToken;
     }
 
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        jsonWriter.writeStringField("access_token", accessToken);
+        jsonWriter.writeStringField("expires_on", expiresOn);
+        jsonWriter.writeStringField("expires_in", expiresIn);
```

Comment:
There is refreshIn field as well, should that be factored in here ?

Reply:
It's not currently a serialized field, and it doesn't actually have any implementation. Maybe it is leftover from a refactor or something? If anything it seems like it should probably just be removed. Thoughts?

Reply:
The value is stored in the parent class and inherited from there. 
We compute and set it there,
If we are serializing the whole object, then in theory that information should be serialized and stored as refresh_in in json.

Reply:
ah of course, will fix.

Reply:
Okay. Here's what I think is true:
1) We never serialize this, so the serialization output doesn't really matter. That said, I've made it match the previous behavior.
2) When we deserialize we only actually need three of the fields to compute everything we need. I've put the test back but we're not reading most of those fields.
3) I's a bigger project than I want to undertake right now but we should _really_ rethink this. Somehow we serialize dates three ways, we store the token twice, etc. I opened an issue to track this (#41661).

I think what I have is correct in that it is the same as what it did before, and for the purposes of finishing the `azure-json` migration I think that's good enough.

Reply:
Why do we need to add serialization logic at this layer? when we don't use it.
User will receive it as AccessToken, AccessToken will have the info at its layer.

Reply:
For now I will leave it as it is as this is the behavior before migrating to azure-json. We can revisit the safety of this when we look at #41661.

Reply:
Removed the extra serialization logic.